src/amo/components/CollectionManager/index.js

```diff
@@ -0,0 +1,176 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { updateCollection } from 'amo/reducers/collections';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import FormOverlay from 'ui/components/FormOverlay';
+import type { CollectionType } from 'amo/reducers/collections';
+import type { UsersStateType } from 'amo/reducers/users';
+import type { ApiStateType } from 'core/reducers/api';
+import type { I18nType } from 'core/types/i18n';
+import type { ElementEvent } from 'core/types/dom';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { DispatchFunc } from 'core/types/redux';
+
+type Props = {|
+  collection: CollectionType | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  siteLang: ?string,
+  siteUserId: number | null,
+|};
+
+type State = {|
+  description?: string | null,
+  name?: string | null,
+|};
+
+export const COLLECTION_OVERLAY = 'COLLECTION_OVERLAY';
+
+export class CollectionManagerBase extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = this.propsToState(props);
+  }
+
+  componentWillReceiveProps(props: Props) {
+    const existingId = this.props.collection && this.props.collection.id;
+    if (props.collection && props.collection.id !== existingId) {
+      // Only reset the form when receiving a collection that the
+      // user is not already editing. This prevents clearing the form
+      // when pressing the submit button.
```

Comment:
This is fine but would the code be a bit easier to follow if we just had and used an `isSubmitting` flag for this? We could render a "submitting" state this way as well and the check could be a bit clearer.

Reply:
This doesn't only apply to submitting, I just used that as an example.